src/Stratis.Bitcoin/TxIndexing/AddressIndexData.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using NBitcoin;
+
+namespace Stratis.Bitcoin.TxIndexing
+{
+    /// <summary>Contains information about operations that happened to an address.</summary>
+    public class AddressIndexData : IBitcoinSerializable
+    {
+        public List<AddressBalanceChange> AddressBalanceChanges;
+
+        /// <inheritdoc />
+        public void ReadWrite(BitcoinStream stream)
+        {
+            if (stream.Serializing)
+            {
+                AddressBalanceChange[] arr = this.AddressBalanceChanges.ToArray();
+
+                stream.ReadWrite(ref arr);
+            }
+            else
+            {
+                AddressBalanceChange[] arr = null;
+                stream.ReadWrite(ref arr);
+
+                this.AddressBalanceChanges = arr.ToList();
+            }
+        }
+    }
+
+    public class AddressBalanceChange : IBitcoinSerializable
+    {
+        /// <summary><c>true</c> if there was a deposit to an address, <c>false</c> if it was a withdrawal.</summary>
+        public bool Deposited;
+
+        public Money Amount;
+
+        /// <summary>Height of a block in which operation was confirmed.</summary>
```

Comment:
*in which the

Also, maybe call it `ConfirmationHeight` or `BalanceChangeHeight` or something?

Reply:
Ok, good idea. noted for next PR.